__tests__/routes/note.test.js

```diff
@@ -0,0 +1,372 @@
+
+afterEach(() => jest.clearAllMocks());
+
+jest.mock('../../api/middleware/authRequired', () =>
+  jest.fn((req, res, next) => next())
+);
+
+jest.mock('../../api/middleware/permissionsRequired', () => ({
+  adminRequired: jest.fn((req, res, next) => next()),
+  superAdminRequired: jest.fn((req, res, next) => next()),
+}));
+
+const app = express();
+app.use(express.json());
+app.use('/notes', notesRouter);
+app.use(handleError);
+describe('Sanity Checks', () => {
+  test('matchers are working', () => {
+    expect(true).toBe(true);
+    expect(20 - 5).toBe(15);
+    expect(9 + 10).not.toEqual(21);
+  });
+
+  test('test environment is being used', () => {
+    expect(process.env.NODE_ENV).toBe('test');
+  });
+});
+
+describe('Notes Router', () => {
+  describe('[GET] /notes', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+    });
+    describe('failed', () => {});
+  });
+
+  describe('[GET] /notes/note_id', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes/1');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+
+      it('responds with non empty object', async () => {
+        const expected = /some text here/i;
+        const actual = res.text;
+        expect(actual).toMatch(expected);
+      });
+    });
+    describe('failed', () => {
+      describe('case - failed 404 error', () => {
+        let res;
+        beforeAll(async () => {
+          res = await request(app).get('/notes/100');
+        });
+
+        it('requires authentication', () => {
+          expect(authRequired).toBeCalled();
+        });
+
+        it('responds with status 200', async () => {
+          const expected = 404;
+          const actual = res.status;
+          expect(actual).toBe(expected);
+        });
+      });
+    });
+  });
+
+  describe('[GET] /notes/mentee/mentee_id', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes/mentee/00ultwqjtqt4VCcS24x6');
+      });
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+
+      it('responds with non empty object', async () => {
```

Comment:
How did we come to the number 5?

Reply:
The seed, for the note table, created five notes for a mentee. The endpoint /notes/mentee/:mentee_id returns a collection of notes, in this case, 5 notes.